pydocx/export/base.py

```diff
@@ -94,14 +106,87 @@ def export(self):
             raise MalformedDocxException
         document = self.main_document_part.document
         if document:
+            # process the document in two passes, since there are some cases
+            # where we can't know what to do until we look at the entire
+            # document (e.g. fields)
+            # In the first pass, discard any generated results
+            self.first_pass = True
+            for result in self.export_node(document):
+                pass
+
+            self._convert_complex_fields_into_simple_fields()
+
+            # actually render the results
+            self.first_pass = False
             for result in self.export_node(document):
                 yield result
 
+    def _convert_complex_fields_into_simple_fields(self):
+        if not self.complex_field_runs:
+            return
+
+        fields = []
+        field = None
+        separate_triggered = False
+        previous_run = None
+
+        runs_to_remove = set()
+
+        for run in self.complex_field_runs:
+            for child in run.children:
+                if field is not None and previous_run is not None:
+                    if previous_run.parent is not run.parent:
+                        # scope has changed
+                        fields.append(field)
+                        field = wordprocessing.SimpleField(instr=field.instr, children=[])
+
+                if isinstance(child, wordprocessing.FieldChar):
+                    separate_triggered = False
+                    runs_to_remove.add(run)
+                    if child.is_type_begin():
+                        field = wordprocessing.SimpleField(children=[])
+                    elif child.is_type_end():
+                        if field is None:
+                            print('end tag without a beginning')
```

Comment:
Do we use print statement for debug logging in other parts of pydocx? Seems like proper logging with loglevel would be nice to have.